src/libtls/tls_crypto.c

```diff
@@ -1195,6 +1200,11 @@ static void build_cipher_suite_list(private_tls_crypto_t *this)
 	{
 		filter_key_suites(this, suites, &count, KEY_ECDSA);
 	}
+	if (!this->eddsa)
+	{
+		filter_key_suites(this, suites, &count, KEY_ED25519);
+		filter_key_suites(this, suites, &count, KEY_ED448);
```

Comment:
This won't work. There are no suites listing these key types. And if we only have an EdDSA key, the suites that could be used (those listing `KEY_ECDSA`) are already filtered at this point. I guess we could change the if statement right above to `if (!this->ecdsa && !this->eddsa)` but that doesn't really have any advantage over the current code that uses the single `ecdsa` flag. So not sure if there actually is any point in separating this further (maybe we could just rename the flag to `ecc`?).